packages/amazonq/test/unit/amazonqFeatureDev/util/files.test.ts

```diff
@@ -56,6 +64,33 @@ describe('file utils', () => {
             assertTelemetry('amazonq_bundleExtensionIgnored', { filenameExt: 'mp4', count: 1 })
         })
 
+        it('prepareRepoData should ignore devfile.yaml when setting is disabled', async function () {
+            const workspace = await getWorkspaceWithDevFile()
+
+            const telemetry = new TelemetryHelper()
+            const result = await prepareRepoData([workspace.uri.fsPath], [workspace], telemetry, {
+                record: () => {},
+            } as unknown as Span<AmazonqCreateUpload>)
+            assert.strictEqual(Buffer.isBuffer(result.zipFileBuffer), true)
+            // checksum is not the same across different test executions because some unique random folder names are generated
+            assert.strictEqual(result.zipFileChecksum.length, 44)
+            assert.strictEqual(telemetry.repositorySize, 12)
+        })
+
+        it('prepareRepoData should ignore devfile.yaml when setting is enabled', async function () {
+            const workspace = await getWorkspaceWithDevFile()
+            await CodeWhispererSettings.instance.updateDevCommandWorkspaceConfigurations(workspace.uri.fsPath, true)
```

Comment:
does this change the actual underlying setting for the user?

Reply:
yeah in the unit test, let me know if there is a concern, should I clean it up?

Reply:
added settings clean up after the test